### PR TITLE
IA-3045: Registry: Pre-select a form in the lower part of the screen as well

### PR DESCRIPTION
**Pull request by beygorghor**
Just like we have on the right hand side, we should also have a form (reference form if available, otherwise any other form) pre selected in the part below the map

Related JIRA tickets : IA-3045
## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [ ] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- fetch org unit type detail to get reference forms of selected type
- pre-select a form 

## How to test

Make sure you have org unit type with a reference form and submission made with this form on an org unit of this type
Visit the parent of this org unit, it should pre-select form selection automatically.

## Print screen / video

https://github.com/BLSQ/iaso/assets/12494624/e2705d3e-3725-4a45-a369-6dfce59e76d9

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/registry/components/Instances.tsx [hunk 1]
```diff
@@ -91,6 +94,29 @@ export const Instances: FunctionComponent<Props> = ({
     const currentForm: Form | undefined = useMemo(() => {
         return formsList?.find(f => `${f.value}` === formIds)?.original;
     }, [formIds, formsList]);
+
+    useEffect(() => {
+        if (
+            formsList?.length &&
+            formsList?.length >= 0 &&
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/registry/components/Instances.tsx]: For same effect, but shorter you can do 
```suggestion
            (formsList?.length ?? 0)>= 0 &&
```

We have the polyfill for `??` now anyway 😁

File: hat/assets/js/apps/Iaso/domains/registry/components/Instances.tsx [hunk 2]
```diff
@@ -91,6 +94,29 @@ export const Instances: FunctionComponent<Props> = ({
...
+            !isFetchingForms &&
+            !formIds &&
+            orgunitTypeDetail
+        ) {
+            let selectedForm = formsList[0].value;
+            if (orgunitTypeDetail.reference_forms.length > 0) {
+                selectedForm = `${orgunitTypeDetail.reference_forms[0].id}`;
+            }
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/registry/components/Instances.tsx]:
```suggestion
         const selectedForm = orgunitTypeDetail.reference_forms.length > 0
             ? `${orgunitTypeDetail.reference_forms[0].id}`
             : formsList[0].value;
```

(You might want to pass that through a proper linter)

File: hat/assets/js/apps/Iaso/domains/registry/components/Instances.tsx [hunk 3]
```diff
@@ -91,6 +94,29 @@ export const Instances: FunctionComponent<Props> = ({
...
+            const newParams = {
+                ...params,
+                formIds: selectedForm,
+            };
+            redirectToReplace(baseUrls.registry, newParams);
+        }
+        // only prselect a form if forms list contain an element and params is empty
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/registry/components/Instances.tsx:116]: Can this not be handled by adding conditions to the `useEffect`  callback?

File: hat/assets/js/apps/Iaso/domains/registry/components/Instances.tsx [hunk 4]
```diff
@@ -50,7 +52,8 @@ export const Instances: FunctionComponent<Props> = ({
         }
         return undefined;
     }, [subOrgUnitTypes, tab]);
-
+    const { data: orgunitTypeDetail, isFetching: isFetchingOrgunitTypeDetail } =
```
Comment (hakifran) [inline hat/assets/js/apps/Iaso/domains/registry/components/Instances.tsx:55]: isFetchingOrgunitTypeDetail declared but never been used

**Review by quang-le — APPROVED** [review]
Just some style suggestions

**Comment by hakifran** [comment]
@beygorghor I have one question, why when I'm on district level I cannot see the forms dropdown on the bottom for HEALTH Facility?
[Screencast from 2024-06-11 11-53-31.webm](https://github.com/BLSQ/iaso/assets/19631540/798f80ef-30da-41a3-ac80-dda8b74edfab)

**Comment by beygorghor** [comment]
> @beygorghor I have one question, why when I'm on district level I cannot see the forms dropdown on the bottom for HEALTH Facility? [Screencast from 2024-06-11 11-53-31.webm](https://github.com/BLSQ/iaso/assets/19631540/798f80ef-30da-41a3-ac80-dda8b74edfab)

Are you sure your org unit type structure is correct? 
HEALTH Facility should be a sub org unit type of the district

File: hat/assets/js/apps/Iaso/domains/registry/components/Instances.tsx (hunk 3 above)
Comment (beygorghor) [reply hat/assets/js/apps/Iaso/domains/registry/components/Instances.tsx:116]: nope

**Review by hakifran — APPROVED** [review]

File: hat/assets/js/apps/Iaso/domains/registry/components/Instances.tsx (hunk 4 above)
Comment (beygorghor) [reply hat/assets/js/apps/Iaso/domains/registry/components/Instances.tsx:55]: yes i removed it after, thx

**Pull request merged**